### DISCUSSION TITLE
How to save useForm, formList from "@mantine/form values in database ?

### DISCUSSION THREAD

**Discussion by Sakkhor909** [mantinedev/mantine#1538] [Q&A] [answered]
I want to save the list array in the database in an onClick event. but form.values is not updated after the `form.addListItem`, `form.removeListItem` or `form.reorderListItem`.
The array gets updated here as I can see here.
`<Code block>{JSON.stringify(form.values, null, 2)}</Code>`
but not in an onClick event immediately
```
   onClick={() => {
               form.addListItem('employees', { name:  '', email: '' })}
              console.log(form.values.employees); // shows previews value
              // want to save the the array in database after form.addListItem
               saveInDatabase(form.values);
              }}
```
It would be great if `form.addListItem()` returns a updated array that I can use.

**Comment by rtivital** [comment, marked as answer]
This is how react state works, it is not possible to return state from `form.addListItem` and other handlers because it will result in bugs with consecutive methods calls:

```ts
form.addListItem('employees', { name:  'a', email: 'a@a' });
form.addListItem('employees', { name:  'b', email: 'b@b' });
```

^^ The result of the second call will not return correct value.

**Reply by Sakkhor909** [reply]
For now, I am muting the array directly before saving it in the database,
for example
```
   // alternative of form.addListItem
   form.values.employees.push(CurrentFormValue);
   saveInDatabase(form.values);

  // alternative of form.removeListItem
  form.values.employees.splice(index, 1);
  saveInDatabase(form.values);
```
Couldn't find any alternative of form.reorderListItem

Now, I know that it's a bad practice. That's why I am seeking help to see if any better way to do this ???

**Reply by rtivital** [reply]
You can subsribe to `form.values.fieldName` in useEffect and run your updates there

**Reply by Sakkhor909** [reply]
But it renders every time I refresh the page or anything related to its UI. At that time, I don't want to call my database API. So I have to also keep track of whether users click on the button or not. only then take action in useEffect.
anyway, thanks @rtivital 
can you tell me an alternative form.reorderListItem?  by muting the array

**Reply by rtivital** [reply]
You can use an array instead of form list and manage all state updates on your side.
